src/applications/pokemon_summary_screen/main.c

```diff
@@ -0,0 +1,2174 @@
+        SetupMoveInfoFromSubscreenButton(summaryScreen);
+    }
+}
+
+static void ChangePage(PokemonSummaryScreen *summaryScreen, s8 delta)
+{
+    s8 page = summaryScreen->page;
+
+    while (TRUE) {
+        page += delta;
+
+        if (page < PSS_PAGE_INFO) {
+            page = PSS_PAGE_EXIT;
+        } else if (page > PSS_PAGE_EXIT) {
+            page = PSS_PAGE_INFO;
+        }
+
+        if (summaryScreen->monData.isEgg == FALSE) {
+            if ((summaryScreen->data->pageFlag & (1 << page)) != 0) {
+                break;
+            }
+        } else {
+            if ((summaryScreen->data->pageFlag & (1 << page)) != 0
+                && (page == PSS_PAGE_MEMO || page == PSS_PAGE_EXIT)) {
+                break;
+            }
+        }
+    }
+
+    if (page == summaryScreen->page) {
+        return;
+    }
+
+    Sound_PlayEffect(SEQ_SE_DP_SELECT5);
+    PokemonSummaryScreen_UpdateSubscreenButtonGfx(summaryScreen);
+    sub_0208FB54(summaryScreen, 0);
+    SetupPageFromSubscreenButton(summaryScreen, page);
+}
+
+static u8 CheckSubscreenPressAndSetButton(PokemonSummaryScreen *summaryScreen)
+{
+    u8 button = PokemonSummaryScreen_CheckSubscreenButtonPress(summaryScreen);
+
+    if (button == PSS_SUBSCREEN_BUTTON_NONE) {
+        return FALSE;
+    }
+
+    summaryScreen->buttonCount = 0;
+    summaryScreen->buttonCurrent = button;
+    summaryScreen->buttonState = 0;
+
+    return TRUE;
+}
+
+static void LoadCurrentPageTilemap(PokemonSummaryScreen *summaryScreen)
+{
+    u32 memberIndex;
+    if (summaryScreen->data->mode == PSS_MODE_SELECT_MOVE && summaryScreen->data->move != MOVE_NONE) {
+        if (summaryScreen->page == PSS_PAGE_BATTLE_MOVES) {
+            memberIndex = 12;
+        } else {
+            memberIndex = 13;
+        }
+    } else {
+        memberIndex = 3 + summaryScreen->page;
+    }
+
+    void *memberBuffer = NARC_AllocAndReadWholeMemberByIndexPair(NARC_INDEX_GRAPHIC__PL_PST_GRA, memberIndex, HEAP_ID_POKEMON_SUMMARY_SCREEN);
+    NNSG2dScreenData *tilemap;
+    NNS_G2dGetUnpackedScreenData(memberBuffer, &tilemap);
+
+    Bg_LoadToTilemapRect(summaryScreen->bgConfig, BG_LAYER_MAIN_3, tilemap->rawData, 0, 0, 32, 32);
+    Bg_ScheduleTilemapTransfer(summaryScreen->bgConfig, BG_LAYER_MAIN_3);
+    Heap_FreeToHeap(memberBuffer);
+
+    if (summaryScreen->page == PSS_PAGE_INFO) {
+        DrawExperienceProgressBar(summaryScreen);
+    } else if (summaryScreen->page == PSS_PAGE_SKILLS) {
+        DrawHealthBar(summaryScreen);
+    }
+}
+
+static void DrawHealthBar(PokemonSummaryScreen *summaryScreen)
+{
+    u16 v0;
+
+    switch (HealthBar_Color(summaryScreen->monData.curHP, summaryScreen->monData.maxHP, 48)) {
+    case BARCOLOR_MAX:
+    case BARCOLOR_GREEN:
+    case BARCOLOR_EMPTY:
+        v0 = 0xc0 | 0xa000;
+        break;
+    case BARCOLOR_YELLOW:
+        v0 = 0xe0 | 0xa000;
+        break;
+    case BARCOLOR_RED:
+        v0 = 0x100 | 0xa000;
+        break;
+    }
```

Comment:
This should just be the palette for a tile, right? `0xA000` for the palette (slot 10) and one of `0x0100`, `0x00E0`, or `0x00C0` for the actual tile number. Looking ahead, it seems like `v0` should thus be the base tile along with its palette (which is what that `17` in `Bg_FillTilemapRect` is for; see the constant definition in `bg_window.h`).

```c
#define TILEMAP_FILL_VAL_INCLUDES_PALETTE 17
```